pkg/resources/provisioning.cattle.io/v1/cluster/validator_test.go

```diff
@@ -131,3 +136,60 @@ func Test_isValidName(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateMachinePoolName(t *testing.T) {
+	tests := []struct {
+		name string
+		fail bool
+	}{
+		{
+			name: strings.Repeat("12345678", 8),
+			fail: true,
+		},
+		{
+			name: strings.Repeat("12345678", 7),
+			fail: false,
+		},
+		{
+			name: "regular-string-test",
+			fail: false,
+		},
+	}
+
+	a := provisioningAdmitter{}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
```suggestion
                for _, tt := range tests {
		    tt := tt
		    t.Run(tt.name, func(t *testing.T) {
	                t.Parallel()
```

Reply:
This allows you to also run the subtests in parallel.

Reply:
Thanks for the code snippet - I wasn't quite sure where to put the parallel piece (above at the top of the test or here).